app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,430 @@
+  }
+
+  private fun configureBottomRightOverlay(spotlightTarget: SpotlightTarget): View {
+    overlayBinding = BottomRightOverlayBinding.inflate(this.layoutInflater)
+    (overlayBinding as BottomRightOverlayBinding).let {
+      it.lifecycleOwner = this
+      it.listener = this
+    }
+
+    (overlayBinding as BottomRightOverlayBinding).customText.text = spotlightTarget.hint
+
+    val arrowParams = (overlayBinding as BottomRightOverlayBinding).arrow.layoutParams
+      as ViewGroup.MarginLayoutParams
+    if (isRtl) {
+      arrowParams.setMargins(
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin),
+        (spotlightTarget.anchorTop.toInt() - getArrowHeight()).toInt(),
+        screenWidth -
+          (spotlightTarget.anchorLeft + getArrowWidth()).toInt(),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin)
+      )
+    } else {
+      arrowParams.setMargins(
+        (spotlightTarget.anchorLeft + spotlightTarget.anchorWidth - getArrowWidth()).toInt(),
+        (spotlightTarget.anchorTop.toInt() - getArrowHeight()).toInt(),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin)
+      )
+    }
+    (overlayBinding as BottomRightOverlayBinding).arrow.layoutParams = arrowParams
+
+    return (overlayBinding as BottomRightOverlayBinding).root
+  }
+
+  private fun configureTopRightOverlay(spotlightTarget: SpotlightTarget): View {
+    overlayBinding = TopRightOverlayBinding.inflate(layoutInflater)
+    (overlayBinding as TopRightOverlayBinding).let {
+      it.lifecycleOwner = this
+      it.listener = this
+    }
+
+    (overlayBinding as TopRightOverlayBinding).customText.text = spotlightTarget.hint
+
+    val arrowParams = (overlayBinding as TopRightOverlayBinding).arrow.layoutParams
+      as ViewGroup.MarginLayoutParams
+    if (isRtl) {
+      arrowParams.setMargins(
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin),
+        (spotlightTarget.anchorTop + spotlightTarget.anchorHeight).toInt(),
+        screenWidth - (spotlightTarget.anchorLeft + getArrowWidth()).toInt(),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin)
+      )
+    } else {
+      arrowParams.setMargins(
+        (spotlightTarget.anchorLeft + spotlightTarget.anchorWidth - getArrowWidth()).toInt(),
+        (spotlightTarget.anchorTop + spotlightTarget.anchorHeight).toInt(),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin),
+        resources.getDimensionPixelSize(R.dimen.spotlight_overlay_arrow_left_margin)
+      )
+    }
+    (overlayBinding as TopRightOverlayBinding).arrow.layoutParams = arrowParams
+
+    return (overlayBinding as TopRightOverlayBinding).root
+  }
+
+  private fun configureTopLeftOverlay(spotlightTarget: SpotlightTarget): View {
```

Comment:
This could probably be consolidated to a single method by adding parameters to the sealed object classes that they can override to specialize configuring the layout for each different case.

Reply:
How might this be done? Though I feel it is better to keep it this way because there isn't really a strong pattern that can help consolidate these 4 layout types (technically 8, since RTL is handled way differently) into 1 and it makes understanding this code much easier.